mpc-core/src/ot/extension/mod.rs

```diff
@@ -202,6 +239,37 @@ mod tests {
         assert_eq!(expected, receive);
     }
 
+    #[rstest]
+    // Test that the cointoss check fails on wrong data
+    fn test_ext_ot_cointoss_failure(mut sender: ExtSenderCore, mut receiver: ExtReceiverCore) {
+        use super::{ExtReceiveCore, ExtSendCore};
+
+        let mut base_sender_setup = receiver.base_setup().unwrap();
+        base_sender_setup.cointoss_commit = [77u8; 32];
+        let base_receiver_setup = sender.base_setup(base_sender_setup).unwrap();
+        let send_seeds = receiver.base_send(base_receiver_setup).unwrap();
+        let err = sender.base_receive(send_seeds).err().unwrap();
+        assert_eq!(err, ExtSenderCoreError::CommitmentCheckFailed);
```

Comment:
More idiomatically, here and in the rest of the tests:
```rust
let res = sender.base_receive(send_seeds);
assert_eq!(res, Err(ExtSenderCoreError::CommitmentCheckFailed)));
```
This will also produce a nicer failure message when `res` is `Ok`